spec/requests/providers/outstanding_mortgage_spec.rb

```diff
@@ -18,40 +18,90 @@
 
-    context 'when an outstanding mortgage value is entered' do
-      context 'with valid values' do
+      context 'when an outstanding mortgage value is entered' do
+        context 'with valid values' do
+          let(:application) { legal_aid_application }
+
+          it 'records the value in the legal aid application table' do
+            legal_aid_application.reload
+            expect(legal_aid_application.outstanding_mortgage_amount).to be_within(0.01).of(321_654.87)
+            expect(legal_aid_application.updated_at.utc.to_i).to be_within(1).of(Time.now.to_i)
+          end
+
+          it 'redirects to the shared ownership page' do
+            expect(response).to redirect_to providers_legal_aid_application_shared_ownership_path(application)
+          end
+        end
+      end
+
+      context 'when an outstanding mortgage value is not entered' do
         let(:application) { legal_aid_application }
+        let(:params) do
+          {
+            legal_aid_application: { outstanding_mortgage_amount: '' },
+            legal_aid_application_id: application.id
+          }
+        end
+
+        it 'shows an error message' do
+          expect(response.body).to include('govuk-error-summary__title')
+        end
 
-        it 'records the value in the legal aid application table' do
+        it 'does not record the value in the legal aid application table' do
           legal_aid_application.reload
-          expect(legal_aid_application.outstanding_mortgage_amount).to be_within(0.01).of(321_654.87)
-          expect(legal_aid_application.updated_at.utc.to_i).to be_within(1).of(Time.now.to_i)
+          expect(legal_aid_application.outstanding_mortgage_amount).to be nil
         end
       end
     end
 
-    context 'when an outstanding mortgage value is not entered' do
-      let(:application) { legal_aid_application }
-      let(:params) do
-        {
-          legal_aid_application: { outstanding_mortgage_amount: '' },
-          legal_aid_application_id: application.id
-        }
+    context 'Submitted with the Save as draft button' do
+      before do
+        params.delete('continue-button')
+        params['draft-button'] = 'Save as draft'
+        patch providers_legal_aid_application_outstanding_mortgage_path(application), params: params
       end
 
-      it 'shows an error message' do
-        expect(response.body).to include('govuk-error-summary__title')
+      context 'when an outstanding mortgage value is entered' do
+        context 'with valid values' do
+          let(:application) { legal_aid_application }
+
+          it 'records the value in the legal aid application table' do
+            legal_aid_application.reload
+            expect(legal_aid_application.outstanding_mortgage_amount).to be_within(0.01).of(321_654.87)
+            expect(legal_aid_application.updated_at.utc.to_i).to be_within(1).of(Time.now.to_i)
```

Comment:
This is old code - just showing up as mine because it's indented inside a new context.  I agree - we don't want to be testing core Active Record stuff in our tests.

Reply:
My experience is that if you don't correct this sort of thing as you spot it, it doesn't get fixed.